### PR TITLE
community[minor]: Added filter search for LanceDB

### PR DESCRIPTION
**Pull request by brnaba-aws**
- [ ] **community**: "vectorstore: added filtering support for LanceDB vector store"

- [ ] **This PR adds filtering capabilities to LanceDB**:
    - **Description:** In LanceDB filtering can be applied when searching for data into the vectorstore. It is using the SQL language as mentioned in the LanceDB documentation.
    - **Issue:** #18235 
    - **Dependencies:** No

- [ ] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [ ] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/lancedb.py [hunk 1]
```diff
@@ -218,14 +218,42 @@ def similarity_search(
...
+                                                    })
         """
         embedding = self._embedding.embed_query(query)  # type: ignore
         tbl = self.get_table(name)
+        filters = kwargs.pop("filter", {})
+        sql_filter = filters.pop("sql_filter", None)
+        prefilter = filters.pop("prefilter", False)
         docs = (
             tbl.search(embedding, vector_column_name=self._vector_key)
+            .where(sql_filter, prefilter=prefilter)
```
Comment (eyurtsev) [inline libs/community/langchain_community/vectorstores/lancedb.py:256]: There's no unit test associated with this change. Could you either add a unit test or help confirm that the default case where sql_filter = None works? Should the query be built conditionally?

Comment (brnaba-aws) [reply libs/community/langchain_community/vectorstores/lancedb.py:256]: if filter is not mentioned, then it works. I tried. That's why I decided to not put any condition around the query.

**Review by eyurtsev — APPROVED** [review]
